lightning/src/routing/router.rs

```diff
@@ -634,8 +634,32 @@ impl PaymentParameters {
 			.expect("PaymentParameters::from_node_id should always initialize the payee as unblinded")
 	}
 
-	/// Includes the payee's features. Errors if the parameters were initialized with blinded payment
-	/// paths.
+	/// Creates parameters for paying to a blinded payee.
+	pub fn blinded(blinded_route_hints: Vec<(BlindedPayInfo, BlindedPath)>) -> Self {
```

Comment:
Another idea is to simply make a `PaymentParameters::for_bolt12_invoice` method that takes a BOLT 12 `Invoice` reference and sets/copies everything (e.g., including features) as needed. We couldn't do this with BOLT 11 invoices because of the crate dependency direction.

Reply:
> Another idea is to simply make a PaymentParameters::for_bolt12_invoice method that takes a BOLT 12 Invoice reference and sets/copies everything (e.g., including features) as needed.

Nice, went with this. 

It would be nice to make the tuple order uniform :/ I don't see how without adding extra allocations or breaking ser. Will raise the issue in the meeting tomorrow.